src/sage/modular/drinfeld_modform/tutorial.py

```diff
@@ -0,0 +1,228 @@
+univariate polynomial ring over `\mathbb{F}_q` and then constructing its
+field of fractions::
+
+    sage: A = GF(3)['T']
+    sage: K.<T> = Frac(A)
+    sage: K
+    Fraction Field of Univariate Polynomial Ring in T over Finite Field of size 3
+    sage: K.base()  # returns A
+    Univariate Polynomial Ring in T over Finite Field of size 3
+
+.. RUBRIC:: Drinfeld period domain
+
+In the classical setting, the domain of any modular forms is the complex
+upper half plane `\mathcal{H}:=\{w\in \mathbb{C} : \mathrm{im}(w)>0\}`.
+The analogue of this plane in the function field setting is the
+*Drinfeld period domain of rank* `r > 1` and it is defined by
+
+.. MATH::
+
+    \Omega^r(\mathbb{C}_{\infty}) :=
+    \mathbb{P}^{r-1}(\mathbb{C}_{\infty})
+    \setminus \{K_{\infty}\text{-rational hyperplanes}\}.
+
+This space is a rigid analytic space and we identify its elements
+with the set of column vectors
+`(w_1,\ldots, w_{r-1}, w_{r})^{\mathrm{T}}` in `\mathbb{C}_{\infty}^r`
+such that the `w_i` are `K_{\infty}`-linearly independant and
+`w_r = \xi`, a nonzero constant in `\mathbb{C}_{\infty}`.
+
+We define a left action of `\mathrm{GL}_r(K_{\infty})` on
+`\Omega^r(\mathbb{C}_{\infty})` by setting
+
+.. MATH::
+
+    \gamma(w) := j(\gamma, w)^{-1}\gamma w
+
+where `j(\gamma, w) := \xi^{-1} \cdot (\text{last entry of }\gamma w)`.
+
+.. RUBRIC:: Universal Drinfeld module over `\Omega^r(\mathbb{C}_{\infty})`
+
+For any `w = (w_1, \ldots, w_{r-1}, \xi)` in
+`\Omega^r(\mathbb{C}_{\infty})` we have a corresponding discrete
+`A`-module which is free of rank `r`:
+
+.. MATH::
+
+    \Lambda^w := Aw_1 \oplus \cdots \oplus Aw_{r-1} \oplus A\xi.
+
+By analytic uniformization, there exists a corresponding Drinfeld module
+
+.. MATH::
+
+    \phi^w : T \mapsto T + g_1(w)\tau + \cdots
+    + g_{r - 1}(w)\tau^{r-1} + g_{r}(w)\tau^{r}.
+
+This Drinfeld module is called the *universal
+Drinfeld `\mathbb{F}_q[T]`-module over `\Omega^r(\mathbb{C}_{\infty})`*
+and its coefficients
+`g_i : \Omega^r(\mathbb{C}_{\infty}) \rightarrow \mathbb{C}_{\infty}`
+are rigid analytic functions satisfying the invariance property:
+
+.. MATH::
+
+    g_i(\gamma(w)) = j(\gamma, w)^{1 - q^i} g_i(w),
+    ~\forall \gamma\in \mathrm{GL}_r(A)
+
+where `g_{r}(w)` never vanishes. Moreover, these coefficients `g_i`
+admits an expansion at infinity, analogous to `q`-expansion principle
+for classical modular forms. The functions `g_i` are known as the
+*coefficients forms at* `T`. More generally, the coefficients of the
+image `\phi^w_a` for any `a\in A` are called the *coefficient forms at*
+`a` and they are an algebraic combination of the coefficient forms at
+`T`.
+
+In the rank two case, the expansion at infinity is of the form
+
+.. MATH::
+
+    g_i(w) = \sum_{i = 0}^{\infty} a_n(g_i)u(w)^i
+
+where `u(w) := e(w)^{-1}` and `e` is the exponential of the Carlitz
+module `\rho:T\mapsto T + \tau`. The analytic parameter `u` is called
+the *parameter at infinity*.
```

Comment:
Added "function"
```suggestion
where `u(w) := e(w)^{-1}` and `e` is the exponential function of the
Carlitz module `\rho:T\mapsto T + \tau`. The analytic parameter `u` is
called the *parameter at infinity*.
```

Also maybe add a reference.